integration-tests/src/test/java/com/datastax/oss/driver/mapper/QueryProviderIT.java

```diff
@@ -107,12 +119,14 @@ public void should_invoke_query_provider() {
 
   public static class FindSliceProvider {
     private final CqlSession session;
+    private final AtomicInteger executionCount;
     private final EntityHelper<SensorReading> sensorReadingHelper;
     private final Select selectStart;
 
     public FindSliceProvider(
         MapperContext context, EntityHelper<SensorReading> sensorReadingHelper) {
       this.session = context.getSession();
+      this.executionCount = ((AtomicInteger) context.getCustomState().get("executionCount"));
```

Comment:
maybe we should add a method on the context that is retrieving state by a key a doing casting to hide that from a client?
`context.getCustomStateByKey("executionCount", AtomicInteger.class)`
it may simplify tests because you do not need to mock the whole custom state but the only value returned by the key